src/main/java/com/google/collegeplanner/servlets/SchedulerServlet.java

```diff
@@ -0,0 +1,143 @@
+    JSONObject body;
+    JSONArray selectedClasses;
+    try {
+      body = getBody(request);
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (NumberFormatException | ClassCastException | ParseException | NullPointerException e) {
+      respondWithError(
+          "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
+      return;
+    }
+
+    prepareLists(selectedClasses, response);
+
+    JSONObject schedules = getSchedules();
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(schedules));
+  }
+
+  /**
+   * Gets the JSON Representation of the body of the POST request
+   */
+  private JSONObject getBody(HttpServletRequest request)
+      throws IOException, ParseException, NullPointerException {
+    String strBody =
+        request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    JSONParser parser = new JSONParser();
+    return (JSONObject) parser.parse(strBody);
+  }
+
+  /**
+   * This method loads the courses and courseList ArrayLists with correct
+   * information. courses is loaded with the lists of sections for each
+   * course while courseList is loaded with the courseID's of the selected
+   * courses.
+   */
+  private void prepareLists(JSONArray classes, HttpServletResponse response) throws IOException {
+    String courseId;
+    URI uri;
+    JSONArray jsonArray;
+    ApiUtil apiUtil = new ApiUtil();
+
+    courseList = new ArrayList<String>();
+    courses = new ArrayList<ArrayList<Section>>();
+    for (Object obj : classes) {
+      courseId = (String) obj;
+      courseList.add(courseId);
+      try {
+        uri = new URI("https://api.umd.io/v1/courses/" + courseId + "/sections");
+      } catch (URISyntaxException e) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+        return;
+      }
+
+      jsonArray = apiUtil.getJsonArray(uri);
+      if (jsonArray == null) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+        return;
+      }
+      courses.add(convertJSONArraytoArrayList(jsonArray));
+    }
+  }
+
+  /**
+   * Returns a JSONArray representing the schedules the
+   * SemesterScheduler returns.
+   */
+  private JSONObject getSchedules() {
+    JSONObject json = new JSONObject();
+    JSONObject temp;
+    JSONArray schedulesJson = new JSONArray();
+    SemesterScheduler scheduler = new SemesterScheduler(courses);
+    ArrayList<Schedule> possibleSchedules = scheduler.getPossibleSchedules();
+
+    for (Schedule schedule : possibleSchedules) {
+      temp = schedule.toJSON();
+      schedulesJson.add(temp);
+    }
+
+    json.put("schedules", schedulesJson);
+
+    return json;
+  }
+
+  /**
+   * Converts a given JSONArray of sections into an ArrayList of
+   * sections.
+   */
+  private ArrayList<Section> convertJSONArraytoArrayList(JSONArray json) {
```

Comment:
Maybe make the function name something more specific to the function of converting sections, and document the parameters